GVRf/Framework/src/org/gearvrf/GVRMesh.java

```diff
@@ -180,9 +177,7 @@ public void setTriangles(char[] triangles) {
      *            Data to bind to the shader attribute.
      */
     public void setFloatVector(String key, float[] floatVector) {
-        isNonNull(floatVector);
-        isValidData(floatVector.length, 1);
-        verticesCheck(floatVector.length, 1);
+        validVectorCheck(key, floatVector, 1);
```

Comment:
I think you should be more consistent in your naming: `validVectorCheck` should probably be `checkValidVector` or (probably better) `checkVectorLength`